lib/rexml/element.rb

```diff
@@ -1035,19 +1121,19 @@ def add_attributes hash
       end
     end
 
-    # Removes an attribute
-    # key::
-    #   either an Attribute or a String.  In either case, the
-    #   attribute is found by matching the attribute name to the argument,
-    #   and then removed.  If no attribute is found, no action is taken.
-    # Returns::
-    #   the attribute removed, or nil if this Element did not contain
-    #   a matching attribute
-    #  e = Element.new('E')
-    #  e.add_attribute( 'name', 'Sean' )             #-> <E name='Sean'/>
-    #  r = e.add_attribute( 'sur:name', 'Russell' )  #-> <E name='Sean' sur:name='Russell'/>
-    #  e.delete_attribute( 'name' )                  #-> <E sur:name='Russell'/>
-    #  e.delete_attribute( r )                       #-> <E/>
+    # :call-seq:
+    #   delete_attribute(name) -> removed_attribute or nil
+    #
+    # Removes a named attribute if it exists;
+    # returns the removed attribute if found, otherwise +nil+:
+    #
+    #   e = REXML::Element.new('foo')
+    #   e.add_attribute('bar', 'baz')
+    #   e.delete_attribute('bar') # => <bar/>
+    #   e.delete_attribute('bar') # => nil
+    #
+    # TODO: Determine whether delete_attribute(attribute) actually works.
```

Comment:
What do we need to do to resolve this?

Reply:
Did not work for me (didn't delete the given REXML::Attribute object).  Resolution is either an example that works or an agreement to omit because it doesn't work.  My personal opinion is that this was not meant to work and is a documentation error.

Reply:
Ah, the example doesn't work as expected?
Could you clarify the expected and the actual? For example, is the example expected or actual?

Reply:
Sorry for this poorly framed issue.  The old doc says that the argument can be "either an Attribute or a String."  An Attribute argument did not work for me:

      e = REXML::Element.new('foo') # => <foo/>
      e.attributes # => {}
      a = REXML::Attribute.new('bar', 'baz') # => bar='baz'
      e.add_attribute(a) # => bar='baz'
      e.attributes # => {"bar"=>bar='baz'}
      e.delete_attribute(a) # => nil
      e.attributes # => {"bar"=>bar='baz'}

Reply:
Confirmed. `e.delete_attribute(a)` doesn't work with the oldest REXML that I have.
We don't support it.